Fix network layout

vandenman opened the pull request:
Some updates to the network module:

- Reintroduces mgm since nobody could reproduce the bug.
- Hides the parallel bootstrap option to avoid firewall issues.
- Fixes the nonzero edges bug.
- Introduces clustering plot and clustering table.
- Slightly reorganized the ui (plots and tables now in separate boxes, enable bootstrap moved to the bootstrap tab).
- Fixed more issues related to Sacha's and Denny's comments
- Included a first unit test (more to come!).

vandenman commented:
@TimKDJ it seems the new unit test for network analysis failed. However, if I run it locally it does pass on both my windows and my linux computer. Did I mess up somewhere?

TimKDJ commented on JASP-Tests/R/tests/testthat/test-networkanalysis.R:
```diff
@@ -0,0 +1,43 @@
+  options$tableClustering <- TRUE
+  options$tableWeightsMatrix <- TRUE
+  options$tableLayout <- TRUE
+  results <- JASPTools::run("NetworkAnalysis", "debug.csv", options, view=FALSE, quiet=TRUE, , sideEffects = "pkgLoading")
```
remove the , in `quiet=TRUE, , sideEffects`

TimKDJ commented on JASP-Tests/R/tests/testthat/test-networkanalysis.R:
```diff
@@ -0,0 +1,43 @@
+# does not test
+# - error handling
+
+test_that("All tables results match", {
```
I'd advise making these into separate tests, but if you want to group them all you should add labels (see below). If you do not do this and a test fails, you will not know which table failed.

vandenman replied:
The reason for one test (with multiple checks) is that running a single network analysis can take some time. And as the tests are not adaptive to run for modified code only, I didn't want to suddenly add > 30 minutes of network tests that are not relevant to anybody but me. If this worry is unnecessary, let me know and I'll separate them.

TimKDJ replied:
I don't think it matters terribly much, it just might be a tad clearer.
Note that the tests can be treated as a normal script.
So this would be possible:

```
context("Network Analysis")

# does not test
# - error handling

options <- JASPTools::analysisOptions("NetworkAnalysis")
options$variables <- c("contNormal", "contcor1", "contcor2")
options$tableCentrality <- TRUE
options$tableClustering <- TRUE
options$tableWeightsMatrix <- TRUE
options$tableLayout <- TRUE
results <- JASPTools::run("NetworkAnalysis", "debug.csv", options, view=FALSE, quiet=TRUE, sideEffects = "pkgLoading")

test_that("General TB table results match", {
  table <- results[["results"]][["generalTB"]][["data"]]
  expect_equal_tables(table,
    list("Network", 3, "2 / 3", 0.333333333333333))
})

test_that("Centrality TB table results match", {
  table <- results[["results"]][["centralityTB"]][["data"]]
  expect_equal_tables(table,
    list("contNormal", -0.577350269189626, -1.12003079401266, -1.14291478283658,
      "contcor1", 1.15470053837925, 0.803219560925303, 0.713968266021615,
      "contcor2", -0.577350269189626, 0.316811233087358, 0.428946516814968))
})
```

vandenman replied:
Okay sounds good, the commit "Separated big unit test into multiple tests" implemented this.

TimKDJ commented on JASP-Tests/R/tests/testthat/test-networkanalysis.R:
```diff
@@ -0,0 +1,43 @@
+
+  table <- results[["results"]][["generalTB"]][["data"]]
+  expect_equal_tables(table,
+    list("Network", 3, "2 / 3", 0.333333333333333)
```
Example of an added label: `list("Network", 3, "2 / 3", 0.333333333333333), label="generalTB"`

TimKDJ commented on JASP-Tests/R/tests/testthat/test-networkanalysis.R:
```diff
@@ -0,0 +1,43 @@
+
+  table <- results[["results"]][["clusteringTB"]][["data"]]
+  expect_equal_tables(table,
+    list("contcor1", 0, 0, 0, 0, "contcor2", 0, 0, 0, 0, "contNormal",
```
I get a different result for this: `list("contNormal", 0, 0, 0, 0, "contcor1", 0, 0, 0, 0, "contcor2",
 0, 0, 0, 0)`

TimKDJ commented on JASP-Tests/R/tests/testthat/test-networkanalysis.R:
```diff
@@ -0,0 +1,43 @@
+
+  table <- results[["results"]][["weightmatrixTB"]][["data"]]
+  expect_equal_tables(table,
+    list(3, 0, 0.0939476582188346, 0, 1, 0.0939476582188346, 0, 0.612057902640958,
```
I get a different result for this: `list(1, 0, 0.0939476582188346, 0, 2, 0.0939476582188346, 0, 0.612057902640958,
 3, 0, 0.612057902640958, 0)`

TimKDJ commented:
Please do a rebase, to fetch my fixes to EFA.

Btw, after you push new commits (or rebase) you can re-start the build on travis.

vandenman commented:
Great feedback Tim! I've rebased and fixed the labeling issue. You don't actually get a different result for the tests, but the result is shown in the order "contNormal", "contcor1", "contcor2", whereas my results are in order "contcor1", "contcor2", "contNormal". I'm not sure what causes this, but at least travis is happy now.

vankesteren commented:
Nice conversation here! Got unittests!
